File: summer_train/fefu/N2.py
with open('input.txt', 'r') as f:
    text = f.readline().strip()
    N = int(f.readline())
    words = [f.readline().strip() for _ in range(N)]   # список хранящий все слова, которые надо проверить

result = []
for word in words:   # с помощью in проверяем каждое слово есть ли оно в исходной строке
    if word in text:
        result.append(1)
    else:
        result.append(0)

with open('output.txt', 'w') as f:
    f.write(' '.join(map(str, result)))

File: 23-24/summer_train/fefu/N2.cpp
#include <iostream>
#include <fstream>
#include <string>
#include <vector>

using namespace std;

int main() {
    ifstream input("input.txt");
    ofstream output("output.txt");

    string text;
    getline(input, text);  // Читаем первую строку (текст)

    int N;
    input >> N;

    vector<string> words(N);   // Вектор для хранения слов
    for (int i = 0; i < N; ++i) {
        input >> words[i]; // читаем слова в вектор
    }

    vector<int> result(N); // Вектор для хранения результата
    for (int i = 0; i < N; ++i) {
        if (text.find(words[i]) != string::npos) { // просто есть ли слово в тексте с помощью встроенной функции 
            result[i] = 1;
        } else {
            result[i] = 0;
        }
    }

    for (int i = 0; i < N; ++i) {
        output << result[i] << " "; // записываем результат
    }

    input.close();
    output.close();

    return 0;
}
